src/OpenLoco/src/Windows/Construction/ConstructionTab.cpp

```diff
@@ -621,14 +621,10 @@ namespace OpenLoco::Ui::Windows::Construction::Construction
     // 0x0049DAF3
     static void disableTrackSlopes(Window* self, TrackObject trackObj, uint64_t disabledWidgets)
     {
-        auto maskedTrackPieces = trackObj.trackPieces & ((1 << 5) | (1 << 8));
-
-        if (maskedTrackPieces != ((1 << 5) | (1 << 8)))
+        if (!trackObj.hasFlags(TrackPieceFlags::slope | TrackPieceFlags::slopedCurve))
             disabledWidgets |= (1 << widx::slope_down) | (1 << widx::slope_up);
 
-        maskedTrackPieces = trackObj.trackPieces & ((1 << 6) | (1 << 8));
-
-        if (maskedTrackPieces != ((1 << 6) | (1 << 8)))
+        if (!trackObj.hasFlags(TrackPieceFlags::steepSlope | TrackPieceFlags::slopedCurve))
```

Comment:
Same here should be two has flags

Reply:
OK, I think I have it fixed on the latest checkin.